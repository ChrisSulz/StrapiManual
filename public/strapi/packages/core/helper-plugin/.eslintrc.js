module.exports = {
  root: true,
  extends: ['custom/front/typescript', 'plugin:storybook/recommended'],
  parserOptions: {
    project: ['./tsconfig.eslint.json'],
  },
  settings: {
    'import/resolver': {
      typescript: {
        project: './tsconfig.eslint.json',
      },
    },
  },
  overrides: [
    {
      files: ['./jest.config.front.js', './webpack.config.js'],
      env: {
        node: true,
      },
      rules: {
        '@typescript-eslint/no-var-requires': 'off',
      },
    },
    {
      files: ['./tests/*'],
      env: {
        jest: true,
      },
    },
  ],
};
