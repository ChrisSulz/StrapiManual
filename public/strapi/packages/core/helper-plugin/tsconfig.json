{
  "extends": "tsconfig/client.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@tests/*": ["./tests/*"]
    }
  },
  "include": ["src", "tests", "jest.config.front.js", "webpack.config.js"],
  "exclude": ["node_modules"]
}
