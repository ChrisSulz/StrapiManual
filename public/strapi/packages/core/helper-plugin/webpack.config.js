const browserslistToEsbuild = require('browserslist-to-esbuild');
const { ESBuildMinifyPlugin } = require('esbuild-loader');
const webpack = require('webpack');

const packageJson = require('./package.json');

const nodeModules = [];
[...Object.keys(packageJson.dependencies), ...Object.keys(packageJson.peerDependencies)].forEach(
  (module) => {
    nodeModules.push(new RegExp(`^${module}(/.+)?$`));
  }
);

/** @type {Omit<import('webpack').Configuration, 'output'>} */
const baseConfig = {
  entry: `${__dirname}/src/index.js`,
  externals: nodeModules,
  mode: process.env.NODE_ENV,
  devtool: process.env.NODE_ENV === 'production' ? false : 'eval-source-map',
  optimization: {
    minimize: process.env.NODE_ENV === 'production',
    minimizer: [
      new ESBuildMinifyPlugin({
        target: browserslistToEsbuild(),
      }),
    ],
  },
  module: {
    rules: [
      {
        test: /\.[jtm]sx?$/,
        use: {
          loader: require.resolve('esbuild-loader'),
          options: {
            loader: 'tsx',
            target: browserslistToEsbuild(),
          },
        },
      },
      {
        test: /\.(png|svg|jpg|gif)$/,
        type: 'asset',
        parser: {
          dataUrlCondition: {
            maxSize: 8192,
          },
        },
      },
    ],
  },
  resolve: {
    extensions: ['.ts', '.js', '.tsx', '.jsx'],
  },
  plugins: [
    new webpack.EnvironmentPlugin({
      NODE_ENV: 'production',
    }),
  ],
};

/** @type {import('webpack').Configuration[]} */
const config = [
  {
    ...baseConfig,
    output: {
      path: `${__dirname}/build`,
      filename: `helper-plugin.esm.js`,
      library: {
        type: 'module',
      },
      environment: { module: true },
    },
    experiments: {
      outputModule: true,
    },
  },
  {
    ...baseConfig,
    output: {
      path: `${__dirname}/build`,
      filename: `helper-plugin.js`,
      library: {
        type: 'commonjs',
      },
    },
  },
];

module.exports = config;
