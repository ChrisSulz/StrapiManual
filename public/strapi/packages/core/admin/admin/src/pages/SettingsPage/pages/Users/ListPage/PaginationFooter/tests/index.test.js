import React from 'react';

import { darkTheme, lightTheme } from '@strapi/design-system';
import { TrackingProvider } from '@strapi/helper-plugin';
import { act, render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { IntlProvider } from 'react-intl';
import { Route, Router } from 'react-router-dom';

import Theme from '../../../../../../../components/Theme';
import ThemeToggleProvider from '../../../../../../../components/ThemeToggleProvider';
import PaginationFooter from '../index';

const makeApp = (history, pagination) => {
  return (
    <TrackingProvider>
      <IntlProvider messages={{}} textComponent="span" locale="en" defaultLocale="en">
        <ThemeToggleProvider themes={{ light: lightTheme, dark: darkTheme }}>
          <Theme>
            <Router history={history}>
              <Route path="/settings/user">
                <PaginationFooter pagination={pagination} />
              </Route>
            </Router>
          </Theme>
        </ThemeToggleProvider>
      </IntlProvider>
    </TrackingProvider>
  );
};

describe('DynamicTable', () => {
  it('renders and matches the snapshot', () => {
    const history = createMemoryHistory();
    const pagination = { pageCount: 2 };
    act(() => history.push('/settings/user?pageSize=10&page=1&sort=firstname'));
    const app = makeApp(history, pagination);

    const { container } = render(app);

    expect(container).toMatchSnapshot();
  });
});
