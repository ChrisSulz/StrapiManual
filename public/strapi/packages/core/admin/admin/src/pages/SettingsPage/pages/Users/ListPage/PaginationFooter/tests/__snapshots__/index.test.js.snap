// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicTable renders and matches the snapshot 1`] = `
.c18 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c10 {
  font-size: 0.875rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c14 {
  font-size: 0.875rem;
  line-height: 1.43;
  color: #666687;
}

.c21 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: revert;
  color: #32324d;
}

.c23 {
  font-size: 0.75rem;
  line-height: 1.33;
  line-height: revert;
  color: #32324d;
}

.c0 {
  padding-top: 16px;
}

.c4 {
  background: #ffffff;
  padding-right: 12px;
  padding-left: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  width: 100%;
  cursor: default;
}

.c7 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
}

.c3 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 16px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 12px;
}

.c15 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 4px;
}

.c6 {
  border: 1px solid #dcdce4;
  min-height: 2rem;
  outline: none;
  box-shadow: 0;
  -webkit-transition-property: border-color,box-shadow,fill;
  transition-property: border-color,box-shadow,fill;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
}

.c6[aria-disabled='true'] {
  color: #666687;
}

.c6:focus-visible {
  outline: none;
}

.c6:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c13 > svg {
  width: 0.375rem;
}

.c13 > svg > path {
  fill: #666687;
}

.c11 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c25[data-state='checked'] .c9 {
  font-weight: bold;
  color: #4945ff;
}

.c16 {
  padding: 12px;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c16:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c16:focus-visible {
  outline: none;
}

.c16:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c19 {
  padding: 12px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  outline: none;
}

.c19:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c19:focus-visible {
  outline: none;
}

.c19:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c20 {
  color: #271fe0;
  background: #ffffff;
}

.c20:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c22 {
  color: #32324d;
}

.c22:hover {
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
}

.c17 {
  font-size: 0.6875rem;
  pointer-events: none;
}

.c17 svg path {
  fill: #c0c0cf;
}

.c17:focus svg path,
.c17:hover svg path {
  fill: #c0c0cf;
}

.c24 {
  font-size: 0.6875rem;
}

.c24 svg path {
  fill: #666687;
}

.c24:focus svg path,
.c24:hover svg path {
  fill: #4a4a6a;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class=""
        >
          <div
            class="c3"
          >
            <div
              aria-autocomplete="none"
              aria-controls="radix-:r3:"
              aria-describedby=":r0:-hint :r0:-error"
              aria-expanded="false"
              aria-label="Entries per page"
              class="c4 c5 c6"
              data-state="closed"
              dir="ltr"
              id=":r0:"
              overflow="hidden"
              role="combobox"
              tabindex="0"
            >
              <span
                class="c7 c8"
              >
                <span
                  class="c9 c10 c11"
                >
                  <span
                    class="c12"
                  >
                    10
                  </span>
                </span>
              </span>
              <span
                class="c8"
              >
                <span
                  aria-hidden="true"
                  class="c13"
                >
                  <svg
                    fill="none"
                    height="1rem"
                    viewBox="0 0 14 8"
                    width="1rem"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M14 .889a.86.86 0 0 1-.26.625L7.615 7.736A.834.834 0 0 1 7 8a.834.834 0 0 1-.615-.264L.26 1.514A.861.861 0 0 1 0 .889c0-.24.087-.45.26-.625A.834.834 0 0 1 .875 0h12.25c.237 0 .442.088.615.264a.86.86 0 0 1 .26.625Z"
                      fill="#32324D"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <span
          class="c9 c14"
        >
          Entries per page
        </span>
      </div>
      <nav
        aria-label="Pagination"
        class=""
      >
        <ol
          class="c15"
        >
          <li>
            <a
              aria-current="page"
              aria-disabled="true"
              class="c16 c17 active"
              href="/settings/user"
              tabindex="-1"
            >
              <div
                class="c18"
              >
                Go to previous page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1rem"
                viewBox="0 0 10 16"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.88 14.12 3.773 8 9.88 1.88 8 0 0 8l8 8 1.88-1.88Z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c19 c20 active"
              href="/settings/user?pageSize=10&page=1&sort=firstname"
            >
              <div
                class="c18"
              >
                Go to page 1
              </div>
              <span
                aria-hidden="true"
                class="c9 c21"
              >
                1
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              class="c16 c22 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c18"
              >
                Go to page 2
              </div>
              <span
                aria-hidden="true"
                class="c9 c23"
              >
                2
              </span>
            </a>
          </li>
          <li>
            <a
              aria-current="page"
              aria-disabled="false"
              class="c16 c24 active"
              href="/settings/user?pageSize=10&page=2&sort=firstname"
            >
              <div
                class="c18"
              >
                Go to next page
              </div>
              <svg
                aria-hidden="true"
                fill="none"
                height="1rem"
                viewBox="0 0 10 16"
                width="1rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 1.88 6.107 8 0 14.12 1.88 16l8-8-8-8L0 1.88Z"
                  fill="#32324D"
                />
              </svg>
            </a>
          </li>
        </ol>
      </nav>
    </div>
  </div>
  <div
    class="c18"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
